tty/pc-tty/ttypc_vtf.c

```diff
@@ -1,617 +1,639 @@
-	virt->G1 = csd_ascii;
-	virt->G2 = csd_supplemental;
-	virt->G3 = csd_supplemental;
-	virt->GL = &virt->G0;
-	virt->GR = &virt->G2;
+	/* Setup tabstops */
+	for (i = 0; i < MAXTABS; i++)
+		vt->tabs[i] = !(i % 8);
 
-	virt->vtsgr = VT_NORMAL;
+	/* Clear escape sequence parameters */
+	_ttypc_vtf_clrparms(vt);
 
-	if (ttypc->color)
-		virt->attr = ((sgr_tab_color[virt->vtsgr]) << 8);
+	/* Reset keyboard modes */
+	vt->ttypc->lockst = 0;
+	if (vt->ttypc->ktype)
+		_ttypc_kbd_updateled(vt->ttypc);
 	else
-		virt->attr = ((sgr_tab_mono[virt->vtsgr]) << 8);
+		_ttypc_bioskbd_updateled(vt->ttypc);
+
+	/* Reset character processing modes */
+	vt->awm = 1;
+	vt->om = 0;
+	vt->ckm = 0;
+	vt->irm = 0;
+	vt->arm = 1;
+	vt->lnm = 0;
+	vt->sc = 0;
+
+	/* Reset character sets */
+	vt->G0 = ascii;
+	vt->G1 = ascii;
+	vt->G2 = supg;
+	vt->G3 = supg;
+	vt->GL = &vt->G0;
+	vt->GR = &vt->G2;
 }
 
 
-/* DECSC - save cursor & attributes */
-void _ttypc_vtf_sc(ttypc_virt_t *virt)
+void _ttypc_vtf_sc(ttypc_vt_t *vt)
 {
-	virt->sc_flag = 1;
-	virt->sc_row = virt->row;
-	virt->sc_col = virt->col;
-	virt->sc_cur_offset = virt->cur_offset;
-	virt->sc_attr = virt->attr;
+	/* Save cursor position */
+	vt->scccol = vt->ccol;
+	vt->sccrow = vt->crow;
+	vt->sccpos = vt->cpos;
+
+	/* Save character sets */
+	vt->scG0 = vt->G0;
+	vt->scG1 = vt->G1;
+	vt->scG2 = vt->G2;
+	vt->scG3 = vt->G3;
+	vt->scGL = vt->GL;
+	vt->scGR = vt->GR;
 
-	virt->sc_awm = virt->m_awm;
+	/* Save VT100 modes & graphic attributes */
+	vt->scawm = vt->awm;
+	vt->scom = vt->om;
+	vt->scsgr = vt->sgr;
+	vt->scattr = vt->attr;
 
-	virt->sc_G0 = virt->G0;
-	virt->sc_G1 = virt->G1;
-	virt->sc_G2 = virt->G2;
-	virt->sc_G3 = virt->G3;
-	virt->sc_GL = virt->GL;
-	virt->sc_GR = virt->GR;
+	vt->sc = 1;
 }
 
 
-/* DECRC - restore cursor & attributes */
-void _ttypc_vtf_rc(ttypc_virt_t *virt)
+void _ttypc_vtf_rc(ttypc_vt_t *vt)
 {
-	if (virt->sc_flag == 1)
+	if (!vt->sc)
 		return;
-	
-	virt->sc_flag = 0;
-	virt->row = virt->sc_row;
-	virt->col = virt->sc_col;
-	virt->cur_offset = virt->sc_cur_offset;
-	virt->attr = virt->sc_attr;
-	
-	virt->m_awm = virt->sc_awm;
 
-	virt->G0 = virt->sc_G0;
-	virt->G1 = virt->sc_G1;
-	virt->G2 = virt->sc_G2;
-	virt->G3 = virt->sc_G3;
-	virt->GL = virt->sc_GL;
-	virt->GR = virt->sc_GR;
+	/* Restore cursor position */
+	vt->ccol = vt->scccol;
+	vt->crow = vt->sccrow;
+	vt->cpos = vt->sccpos;
+
+	/* Restore character set */
+	vt->G0 = vt->scG0;
+	vt->G1 = vt->scG1;
+	vt->G2 = vt->scG2;
+	vt->G3 = vt->scG3;
+	vt->GL = vt->scGL;
+	vt->GR = vt->scGR;
+
+	/* Restore VT100 modes & graphic attributes */
+	vt->awm = vt->scawm;
+	vt->om = vt->scom;
+	vt->sgr = vt->scsgr;
+	vt->attr = vt->scattr;
+
+	vt->sc = 0;
 }
 
 
-/* device attributes */
-void _ttypc_vtf_da(ttypc_virt_t *virt)
+void _ttypc_vtf_da(ttypc_vt_t *vt)
 {
-#if 0
-	static uint8_t *response = (u_char *)DA_VT220;
+	/* 62 - class 2 terminal, c - end of attributes list */
+	static char *da = "\033[62;c";
+	write(0, da, sizeof(da));
```

Comment:
`sizeof(da) = 4` (`sizeof(char *)`). It should be `static const char da[] = "\033[62;c";`

Reply:
Fixed.